zapcore/write_syncer.go

```diff
@@ -83,6 +84,62 @@ func (w writerWrapper) Sync() error {
 	return nil
 }
 
+type bufferedWriter struct {
+	size int // in bytes
+	buf  []byte
+	ws   WriteSyncer
+}
+
+// Buffer creates a WriteSyncer that buffers the specified number of bytes in
+// memory. Buffered WriteSyncers aren't safe for concurrent use, and should be
+// wrapped by Lock in production.
+func Buffer(size int, ws WriteSyncer) WriteSyncer {
+	return &bufferedWriter{
+		size: size,
+		buf:  buffers.Get(),
+		ws:   ws,
+	}
+}
+
+func (b *bufferedWriter) Write(bs []byte) (int, error) {
+	// Attempt to buffer.
+	if len(b.buf)+len(bs) <= b.size {
```

Comment:
Really? That seems like an off-by-one error.

If you told me that you could hold four apples but always drop the fourth apple, I'd call you a liar ;)